performance_decomposition/apps_scripts/handler.gs

```diff
@@ -0,0 +1,223 @@
+/*
+    Name:        WeDiscover - Performance Decomposition, Google Apps Script
+
+    Description: A handler class for running core processes.
+
+    License:     https://github.com/we-discover/public/blob/master/LICENSE
+
+    Version:     1.0.1
+
+    Released:    2022-08-01
+
+    Contact:     scripts@we-discover.com
+*/
+
+
+class DecompHandler {
+
+  constructor(resetMode) {
+    this.workbook = SpreadsheetApp.getActiveSpreadsheet();
+    this.sheet = this.workbook.getSheetByName(decompositionSheetName);
+
+    this.workbook.toast("Running background process.", "Processing...", 2);
+    Utilities.sleep(1000);
+  }
+
+  /**
+   * Calculate the comparison period multiplier. 
+   * 1 = Previous, 2 = lagged 1, 3 = lagged 2.
+  */
+  _getComparisonPeriodMultiplier(comparisonType) {
+    var multiplier = 1;
+    var re = new RegExp('Previous Period \\(-(.*)\\)');
+    var matches = re.exec(comparisonType);
```

Comment:
💡Instead of using `new RegExp(...)`, you can enclose regex patterns in a pair of `/`s. So lines 32 and 33 could be combined into:

`const matches = /Previous Period \\(-(.*)\\)/.exec(comparisonType);`